packages/trail-fastify-server/lib/index.js

```diff
@@ -27,6 +27,9 @@ module.exports = async function (options) {
     if (settings.use.graphQL) {
       server.register(require('@nearform/trail-fastify-graphql-plugin'), { db: settings.db })
     }
+    if (settings.use.corsPlugin) {
+      server.register(require('fastify-cors'), { })
+    }
```

Comment:
let's always register the plugin with an origin setting that comes from the corsOrigin option. If that is set to false (which we can keep as the default as you already did) it will disable CORS, as explained in the docs.

Reply:
The current implementation of loadSettings does not allow the values to be of multiple type. Hence if the default value is given as boolean, it will convert the override value to a boolean too.
Will change the settings to:
```
     corsPlugin: resolve('corsOrigin', 'TRAIL_CORS_ORIGIN', '')
```
And move the settings under `http` section of settings